opencog/atoms/value/PtrValue.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * opencog/atoms/value/PtrValue.h
+ *
+ * Copyright (C) 2018 Vitaly Bogdanov
+ * All Rights Reserved
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _OPENCOG_PTR_VALUE_H
+#define _OPENCOG_PTR_VALUE_H
+
+#include <opencog/atoms/value/Value.h>
+#include <opencog/atoms/atom_types/atom_types.h>
+
+namespace opencog
+{
+
+/** \addtogroup grp_atomspace
+ *  @{
+ */
+
+/**
+ * PtrValue holds an pointer to the native object. Typical usecase is passing
+ * object to the GroundedSchemaNode for processing.
+ */
+class PtrValue
+	: public Value
+{
+protected:
+	std::shared_ptr<void> ptr;
+
+public:
+	/** Function to be called when value is destroyed to release the pointer */
+	using Deleter = std::function<void(void*)>;
+
+	PtrValue(void* ptr, Deleter deleter)
```

Comment:
Maybe `deleter` could have `noop` (as defined in PtrValueUTest.cxxtest), as default argument? No sure whether `createPtrValue` can deal with that though...